### ISSUE TITLE
CalciteNotice - make the floating look opt-in

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-components#3426]
### Description
Notice sometimes needs to be on a white background, and its box-shadow doesn't always work.
Also, in context with other UI language, the box-shadow may actually be a distraction.

Suggest
* only showing box-shadow when a `floating` prop is added.
* in default add a 1px border-color-3
* also consider remove the accent border for the "neutral" color.
* probably should deprecate `color` and replace it with `status` but that's probably another issue

### Acceptance Criteria 
Design should work on white backgrounds.
Design should match surrounding UI.
When `floating` is true, the design is as it is now.

#### Which Component
Notice

#### Example Use Case
On white backgrounds.

#### Current
![Screen Shot 2021-11-03 at 4 35 37 PM](https://user-images.githubusercontent.com/12503298/140233662-58db1581-ede4-4c36-96f3-919c9d916f35.png)

Proposed (shown on white bg)
![Screen Shot 2021-11-03 at 4 34 52 PM](https://user-images.githubusercontent.com/12503298/140233687-7e367a02-76a4-479b-9f4b-78bc244d45be.png)

#### Proposed (shown on bg color)
![Screen Shot 2021-11-03 at 4 34 18 PM](https://user-images.githubusercontent.com/12503298/140233719-2da0bb65-0ebd-4d90-b909-62fe51729341.png)

**Comment by asangma** [comment]
@macandcheese Would dig getting your thoughts on this.

**Comment by macandcheese** [comment]
Some thoughts: 

The proposed design overlaps with the `calcite-tip` component - and invites fragmentation of notice use across products. I know the tip is structured differently with slots, but visually they end up almost the same. 

For the third screenshot - wouldn't the current notice design with shadow work since it's on a grey bg - perhaps preferential there (folks can also change the white background to be foreground-1 via css variable).

That said, it's been requested before, and there is some precedent for this kind of adjustment - on color picker we have "minimal" that removes the border, and on date picker we (used to have?) a prop to basically do this.

If we *were* to make these changes, the default appearance should remain the same, and anything new should be "opt-in" via a prop. 

Here's how we use this across the developer site - adding a border here is a non-starter since these exist inside already very border-heavy UI. 

![Screen Shot 2021-11-04 at 10 25 18 AM](https://user-images.githubusercontent.com/4733155/140389555-d0dc7d36-a1a7-48ce-b329-28190ef1116c.png)

![Screen Shot 2021-11-04 at 10 35 46 AM](https://user-images.githubusercontent.com/4733155/140390559-9ff078b0-4255-4429-9cd1-f768d56f9669.png)


To the other points:

We don't currently have neutral, but I think adding that and changing color / status / etc., are proposed and encompassed with other issues, so I agree those would be nice to add and improve the component 👍

**Comment by macandcheese** [comment]
Yep - that should be within the scope of the linked issue. The updated "solid", "outline-fill", and "transparent" appearance types will be further customizable with tokens for corner radius, box shadow, etc. I've added it explicitly to the issue description to ensure they are added.
